zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -180,7 +180,7 @@ public static void setPreallocSize(long size) {
      * @param serverStats used to update fsyncThresholdExceedCount
      */
     @Override
-    public void setServerStats(ServerStats serverStats) {
+    public synchronized void setServerStats(ServerStats serverStats) {
```

Comment:
I'm not sure if we need to synchronize here, but it shouldn't be part of this patch.

Reply:
This is spotbugs error:
`[ERROR] Inconsistent synchronization of org.apache.zookeeper.server.persistence.FileTxnLog.serverStats; locked 66% of time [org.apache.zookeeper.server.persistence.FileTxnLog, org.apache.zookeeper.server.persistence.FileTxnLog, org.apache.zookeeper.server.persistence.FileTxnLog] Unsynchronized access at FileTxnLog.java:[line 184]Synchronized access at FileTxnLog.java:[line 381]Synchronized access at FileTxnLog.java:[line 382] IS2_INCONSISTENT_SYNC
`

I think this change is safe

Reply:
Ok, let's keep it then. Looks safe to me too.